samples/extensions/descriptor_buffer_basic/descriptor_buffer_basic.cpp

```diff
@@ -0,0 +1,471 @@
+
+		vkCmdBindPipeline(draw_cmd_buffers[i], VK_PIPELINE_BIND_POINT_GRAPHICS, pipeline);
+
+		VkViewport viewport = vkb::initializers::viewport(static_cast<float>(width), static_cast<float>(height), 0.0f, 1.0f);
+		vkCmdSetViewport(draw_cmd_buffers[i], 0, 1, &viewport);
+
+		VkRect2D scissor = vkb::initializers::rect2D(width, height, 0, 0);
+		vkCmdSetScissor(draw_cmd_buffers[i], 0, 1, &scissor);
+
+		const auto &vertex_buffer = models.cube->vertex_buffers.at("vertex_buffer");
+		auto       &index_buffer  = models.cube->index_buffer;
+
+		VkDeviceSize offsets[1] = {0};
+		vkCmdBindVertexBuffers(draw_cmd_buffers[i], 0, 1, vertex_buffer.get(), offsets);
+		vkCmdBindIndexBuffer(draw_cmd_buffers[i], index_buffer->get_handle(), 0, models.cube->index_type);
+
+		// Descriptor buffer bindings
+		// Binding 0 = uniform buffer
+		VkDescriptorBufferBindingInfoEXT descriptor_buffer_binding_info[2]{};
+		descriptor_buffer_binding_info[0].sType   = VK_STRUCTURE_TYPE_DESCRIPTOR_BUFFER_BINDING_INFO_EXT;
+		descriptor_buffer_binding_info[0].address = resource_descriptor_buffer->get_device_address();
+		descriptor_buffer_binding_info[0].usage   = VK_BUFFER_USAGE_RESOURCE_DESCRIPTOR_BUFFER_BIT_EXT;
+		// Binding 1 = Image
+		descriptor_buffer_binding_info[1].sType   = VK_STRUCTURE_TYPE_DESCRIPTOR_BUFFER_BINDING_INFO_EXT;
+		descriptor_buffer_binding_info[1].pNext   = nullptr;
+		descriptor_buffer_binding_info[1].address = image_descriptor_buffer->get_device_address();
+		descriptor_buffer_binding_info[1].usage   = VK_BUFFER_USAGE_SAMPLER_DESCRIPTOR_BUFFER_BIT_EXT | VK_BUFFER_USAGE_RESOURCE_DESCRIPTOR_BUFFER_BIT_EXT;
+		vkCmdBindDescriptorBuffersEXT(draw_cmd_buffers[i], 2, descriptor_buffer_binding_info);
+
+		uint32_t     buffer_index_ubo = 0;
+		VkDeviceSize alignment        = descriptor_buffer_properties.descriptorBufferOffsetAlignment;
+		VkDeviceSize buffer_offset    = 0;
+
+		// Global Matrices (set 0)
+		vkCmdSetDescriptorBufferOffsetsEXT(draw_cmd_buffers[i], VK_PIPELINE_BIND_POINT_GRAPHICS, pipeline_layout, 0, 1, &buffer_index_ubo, &buffer_offset);
+
+		// Set and offset into descriptor for each model
+		for (uint32_t j = 0; j < static_cast<uint32_t>(cubes.size()); j++)
+		{
+			// Uniform buffer (set 1)
+			// Model ubos start at offset * 1 (slot 0 is global matrices)
+			buffer_offset = alignment + j * alignment;
+			vkCmdSetDescriptorBufferOffsetsEXT(draw_cmd_buffers[i], VK_PIPELINE_BIND_POINT_GRAPHICS, pipeline_layout, 1, 1, &buffer_index_ubo, &buffer_offset);
+			// Image (set 2)
+			uint32_t buffer_index_image = 1;
+			buffer_offset               = j * alignment;
+			vkCmdSetDescriptorBufferOffsetsEXT(draw_cmd_buffers[i], VK_PIPELINE_BIND_POINT_GRAPHICS, pipeline_layout, 2, 1, &buffer_index_image, &buffer_offset);
+			draw_model(models.cube, draw_cmd_buffers[i]);
+		}
+
+		// @todo: can't mix descriptor buffers and descriptors, so we disable the UI for now
+		// draw_ui(draw_cmd_buffers[i]);
```

Comment:
What issues do you get with the ui? I don't get any.

Reply:
When I started writing the sample, mixing draw calls with and without descriptor buffers caused validation errors. With the latest SDK they no longer appear, so I guess it was an error in the validation.

Will re-enable the UI now.